routes/monitor.js

```diff
@@ -12,24 +12,321 @@ router.get('/', ensureAuthenticated, function(req, res) {
+        int_time_averaging_window == false
+    ){
+        console.log("something is zero")
+        res.send("false");
+        return(404);
+    } else {
+        console.log("oK")
+    }
+    
+    
+    var json_project_pmts = {};
+    var json_zero_pmts    = {};
+    for(var i = 0; i < json_pmts.length;i += 1){
+        var str_channel_pmt  = "channels."+String(json_pmts[i]);
+        var str_channel_pmt2 = "$" + str_channel_pmt
+        json_project_pmts[String(json_pmts[i])] = 1;
+        json_zero_pmts[str_channel_pmt] = {$ifNull: [ str_channel_pmt2, -1 ] };
+    
+    }
+    
+    
+    
+    var mongo_pipeline = [];
+    // roughly filter data to relevant hosts and timeframe
+    mongo_pipeline.push(
+    {"$match":{
+        "_id":{"$gt": oid_time_start,"$lt": oid_time_end},
+        "host":{$regex:"reader[0-2]_reader_0", $options:"$i" }}
+    });
+    
+    // create time in seconds and kick out unnecesarry pmts
+    mongo_pipeline.push({
+        '$project': {
+            time:{
+                '$divide':[
+                    {'$subtract': [ {"$toDate":"$_id"}, new Date(int_time_start*1000) ]},
+                    1000
+                ]
+            },
+            "_id": 0,
+            channels: json_project_pmts,
+        },
+    },{
+        '$group':{
+            _id: "$time",
+            channels: {"$mergeObjects": "$channels"}
+        }
+    });
+    
+    // complicateed setup to generate propper json to check if a rate is not set
+    // set it to -1 so max one step later does not ignore thos timebins
+    $project =  {
+        timebin:{
+            '$trunc': [
+                {'$divide': [
+                    "$_id",
+                    int_time_averaging_window
+                ]}
+            ],
+        }
+    }
+    for(var key in json_zero_pmts){
+        value = json_zero_pmts[key];
+        $project[key] = value;
+    }
+    
+    
+    
+    // merge into timebins
+    mongo_pipeline.push({
+        $project
+    })
+    
+    if(true){
+        mongo_pipeline.push({
+            '$group':{
+                _id: {
+                    '$add':[
+                        {'$multiply': [
+                            "$timebin",
+                            int_time_averaging_window
+                        ]},
+                        int_time_start
+                    ]
+                },
+                channels: {"$max": "$channels"},
+            }
+        })
+    };
+    // and sort by id again
+    mongo_pipeline.push({
+        '$sort': {
+            "_id": 1,
+        }
     });
+    
+    //console.log(mongo_pipeline);
+    //res.json(mongo_pipeline);
+    
+    collection.aggregate(
+        mongo_pipeline,
+        function(e,docs){
+            res.json(docs);
+        }
+    );
+    
 });
 
+
+
+// returns cable_map (used to generate new tpc svg)
+// no authentintification as python will need to access these functions
+router.get('/get_cable_map', function(req,res){
```

Comment:
No unauthenticated routes. If you need to regenerate the svg then run the script from somewhere with database access like the gateway (or old-gw)